docs/DafnyRef/Statements.md

```diff
@@ -1537,7 +1537,14 @@ much as lemmas might be used in mathematical proofs.
 
 Using `...` as the argument of the statement is part of module refinement, as described in [Section 22](#sec-module-refinement).
 
-TO BE WRITTEN - assert by statements
+In the `by` form of the `assert` statement, there is an additional block of statements.
+Those statements are typically a sequence of [lemmas](#sec-lemmas) or a [calc statement](#sec-calc-statement).
```

Comment:
Also, I found it typical to assert other assertions (when refactoring proofs), ghost if statements, and reveal statements.

Also, could you give an example of an assert..by statement?